i3ipc.py

```diff
@@ -413,6 +413,15 @@ def __init__(self, data, parent, conn):
             else:
                 setattr(self, attr, None)
 
+        # Make sure marks is never None (use []) to simplify client code. Older
+        # versions of i3 used mark instead of marks, so collapse that into
+        # marks to provide a consistent API.
+        if not self.marks:
```

Comment:
Thanks for the git-undo tip. That's nice.

Yes, I realized I should be doing pull requests in a topic branch once I
discovered that they just slurp in everything from the branch they were
created on. :-)

Do you want me to re-do this pull request on a topic branch, or is there
some way to change the branch that a pull request is tracking, or are you
okay with this one being on master?


On Fri, Apr 22, 2016 at 5:41 PM, Tony Crisci <notifications@github.com>
wrote:

> In i3ipc.py
> <https://github.com/acrisci/i3ipc-python/pull/33#discussion_r60819086>:
>
> > @@ -413,6 +413,15 @@ def __init__(self, data, parent, conn):
> >              else:
> >                  setattr(self, attr, None)
> >
> > +        # Make sure marks is never None (use []) to simplify client code. Older
> > +        # versions of i3 used mark instead of marks, so collapse that into
> > +        # marks to provide a consistent API.
> > +        if not self.marks:
>
> Yes, that would be good if you removed it. I like git undo
> <https://github.com/tj/git-extras/blob/master/Commands.md#git-undo> from
> git-extras for that.
>
> Also, you should be doing pull requests in a topic branch not the master
> branch.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/acrisci/i3ipc-python/pull/33/files/2ea9f619a2d16eab83b6f7e81933a7c7449fd338#r60819086>
>